string room
---
bool has_reached